azure-pipelines.yml

```diff
@@ -205,6 +207,11 @@ jobs:
       TargetPath: '$(Build.ArtifactStagingDirectory)/metadata'
       artifactName: metadata
 
+  - task: PipAuthenticate@1
+    condition: eq(variables['Build.SourceBranch'], 'refs/heads/release')
+    displayName: 'Pip Authenticate'
+    inputs:
+      artifactFeeds: ${{ variables.AZURE_ARTIFACTS_FEEDS }}
```

Comment:
Other tasks use [macro syntax](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#macro-syntax-variables) as `$(variables.AZURE_ARTIFACTS_FEEDS)`. Does [template expression syntax](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#template-expression-syntax) work here

Reply:
You're right, the variable defined in ADO UI is empty when use `${{ variables.AZURE_ARTIFACTS_FEEDS }}`. I've changed to `$(AZURE_ARTIFACTS_FEEDS)`